### PR TITLE
add Keyboard Rartlite and support via

### PR DESCRIPTION
**Pull request by labahuy**
## Description

add new keyboard rartlite and support keymap via

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/rart/rartlite/rartlite.c [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    switch (get_highest_layer(state)) {
+      case 1:
+        writePinHigh(B1);
+        break;
+      default: //  for any other layers, or the default layer
+        writePinLow(B1);
+        break;
+      }
+    return layer_state_set_user(state);
+}
```
Comment (drashna) [inline keyboards/rart/rartlite/rartlite.c:44]:
```suggestion
void keyboard_pre_init_user(void) {
  setPinOutput(B1);
}

layer_state_t layer_state_set_kb(layer_state_t state) {
    writePin(B1, layer_state_cmp(state, 1));
    return layer_state_set_user(state);
}
```

The rest can be replaced with the following in your config.h:
```c
#define LED_NUM_LOCK_PIN F5
#define LED_NUM_LOCK_PIN F6
```

And you may or may not need `#define LED_PIN_ON_STATE 0`

File: keyboards/rart/rartlite/readme.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+* Keyboard Maintainer: [Alabahuy](https://github.com/alabahuy)
+* Hardware Supported: RARTLITE PCB, Promicro, Mini USB
+* Hardware Availability: Private GB
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rart/rartlite:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```
Comment (drashna) [inline keyboards/rart/rartlite/readme.md:14]: Could you include instructions on how to enter the bootloader?

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlite/readme.md (hunk 2 above)
Comment (labahuy) [reply keyboards/rart/rartlite/readme.md:18]: Pro micro doesn't need to add another bootloader right?

File: keyboards/rart/rartlite/rartlite.c [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rartlite.h"
+
+void keyboard_pre_init_user(void) {
+  setPinOutput(B1);
+}
```
Comment (zvecr) [inline keyboards/rart/rartlite/rartlite.c:19]:
```suggestion
void keyboard_pre_init_kb(void) {
  setPinOutput(B1);
  
  keyboard_pre_init_user();
}
```

File: keyboards/rart/rartlite/rules.mk [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
+ENCODER_ENABLE = no
+
+LAYOUTS = right_numpad left_numpad
```
Comment (zvecr) [inline keyboards/rart/rartlite/rules.mk:25]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlite/rartlite.c (hunk 3 above)
Comment (labahuy) [reply keyboards/rart/rartlite/rartlite.c]: done :)

File: keyboards/rart/rartlite/rules.mk (hunk 4 above)
Comment (labahuy) [reply keyboards/rart/rartlite/rules.mk]: done :)

File: keyboards/rart/rartlite/readme.md (hunk 2 above)
Comment (drashna) [reply keyboards/rart/rartlite/readme.md:18]: Nope.  But what I mean is how to enter the bootloader so you can flash the firmware.  Eg, does it support bootmagic, is there a dedicated hardware reset button, etc.

Comment (labahuy) [reply keyboards/rart/rartlite/readme.md:22]: ok, done :)

**Review by drashna — APPROVED** [review]

File: keyboards/rart/rartlite/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        _______, _______,          _______,          _______,          _______,          _______, _______, _______,   _______, _______, _______
+    ),
+
+    [3] = LAYOUT_right_numpad(
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,   _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,   _______, _______, _______,
+        _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,          _______,   _______, _______, _______,
+        _______, _______,          _______,          _______,          _______,          _______, _______, _______,   _______, _______, _______
+    )
+
```
Comment (fauxpark) [inline keyboards/rart/rartlite/keymaps/default/keymap.c:49]:
```suggestion
```

File: keyboards/rart/rartlite/readme.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+* Hardware Supported: RARTLITE PCB, Promicro, Mini USB
+* Hardware Availability: Private GB
+
+Make example for this keyboard (after setting up your build environment):
+
+    make rart/rartlite:default
+    
+Flashing example for this keyboard :
+
+    make rart/rartlite:default:avrdude
```
Comment (fauxpark) [inline keyboards/rart/rartlite/readme.md:16]:
```suggestion
Flashing example for this keyboard:

    make rart/rartlite:default:flash
```

File: keyboards/rart/rartlite/readme.md [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    
+## Bootloader :
```
Comment (fauxpark) [inline keyboards/rart/rartlite/readme.md:18]:
```suggestion
## Bootloader
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/rart/rartlite/readme.md (hunk 7 above)
Comment (labahuy) [reply keyboards/rart/rartlite/readme.md]: Done :)

**Comment by labahuy** [comment]
any update @zvecr @fauxpark ??

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
For future reference, we recommend against committing to your `master` branch as you've done here, because pull requests from modified `master` branches can make it more difficult to keep your QMK fork updated. It is highly recommended for QMK development – regardless of what is being done or where – to keep your master updated, but **NEVER** commit to it. Instead, do all your changes in a branch (branches are basically free in Git) and issue PRs from your branches when you're developing.

There are instructions on how to keep your fork updated here:

[**Best Practices: Your Fork's Master: Update Often, Commit Never**](https://docs.qmk.fm/#/newbs_git_using_your_master_branch)

[Fixing Your Branch](https://docs.qmk.fm/#/newbs_git_resynchronize_a_branch) will walk you through fixing up your `master` branch moving forward. If you need any help with this just ask.

Thanks for contributing!

**Pull request merged**